controller/k8s/clientset.go

```diff
@@ -21,3 +22,16 @@ func NewSpClientSet(kubeConfig string) (*spclient.Clientset, error) {
 	config.WrapTransport = prometheus.ClientWithTelemetry("sp", wt)
 	return spclient.NewForConfig(config)
 }
+
+// NewTsClientSet returns a Kubernetes TrafficSplit client for the given
+// configuration.
+func NewTsClientSet(kubeConfig string) (*tsclient.Clientset, error) {
```

Comment:
I think we can extract the common code from the two functions in this file

Reply:
Hilariously, because error handling has to be repeated at every call layer, this refactor actually makes the code *longer*: 

```
--- a/controller/k8s/clientset.go
+++ b/controller/k8s/clientset.go
@@ -5,33 +5,41 @@ import (
        spclient "github.com/linkerd/linkerd2/controller/gen/client/clientset/versioned"
        "github.com/linkerd/linkerd2/pkg/k8s"
        "github.com/linkerd/linkerd2/pkg/prometheus"
+       "k8s.io/client-go/rest"

        // Load all the auth plugins for the cloud providers.
        _ "k8s.io/client-go/plugin/pkg/client/auth"
 )

+func newConfig(kubeConfig string, telemetryName string) (*rest.Config, error) {
+       config, err := k8s.GetConfig(kubeConfig, "")
+       if err != nil {
+               return nil, err
+       }
+
+       wt := config.WrapTransport
+       config.WrapTransport = prometheus.ClientWithTelemetry(telemetryName, wt)
+       return config, nil
+}
+
 // NewSpClientSet returns a Kubernetes ServiceProfile client for the given
 // configuration.
 func NewSpClientSet(kubeConfig string) (*spclient.Clientset, error) {
-       config, err := k8s.GetConfig(kubeConfig, "")
+       config, err := newConfig(kubeConfig, "sp")
        if err != nil {
                return nil, err
        }

-       wt := config.WrapTransport
-       config.WrapTransport = prometheus.ClientWithTelemetry("sp", wt)
        return spclient.NewForConfig(config)
```

I still think it's a good idea though.